File: Exercism/c/armstrong-numbers/src/armstrong_numbers.c
#include "armstrong_numbers.h"
#include <math.h>
#include <stdio.h>


int is_armstrong_number(int candidate) {
  int len = 0;
  for (int n = candidate ; n != 0 ; n /= 10)
    len++;

  int sum = 0;
  for (int n = candidate ; n != 0 ; n /= 10)
    sum += pow(n%10, len);

  return sum == candidate;
}

File: Exercism/dart/armstrong-numbers/lib/armstrong_numbers.dart
import 'dart:math';

class ArmstrongNumbers {
  bool isArmstrongNumber(int candidate) {
    String number = candidate.toString();

    int sum = number.split('')
    .fold(0, (int sum, String x) => sum + pow(int.parse(x), number.length) as int);

    return sum == candidate;
  }
}
